SoftLayer/CLI/block/detail.py

```diff
@@ -5,16 +5,35 @@
 import SoftLayer
 from SoftLayer.CLI import environment
 from SoftLayer.CLI import formatting
+from SoftLayer.CLI import helpers
 from SoftLayer import utils
 
 
+def get_block_volume_id(volume_id, block_manager):
+    """Returns the volume id.
+
+    :param volume_id: ID of volume.
+    :param block_manager: Block Storage Manager.
+    :return: Returns the volume id.
+    """
+    storage_list = block_manager.list_block_volumes()
```

Comment:
`list_block_volumes()` takes in a username parameter to filter by that, which will let you only get the volume back that matches the username. Also pass in a mask of just `id` help reduce API call returned data.